planner/core/planbuilder.go

```diff
@@ -374,7 +374,7 @@ func NewPlanBuilder(sctx sessionctx.Context, is infoschema.InfoSchema, processor
 
 // Build builds the ast node to a Plan.
 func (b *PlanBuilder) Build(ctx context.Context, node ast.Node) (Plan, error) {
-	b.optFlag = flagPrunColumns
+	b.optFlag = flagPrunColumns | flagPrunColumnsAgain
```

Comment:
Can we move `flagPrunColumnsAgain` to where assigned `flagPredicatePushDown`?

Reply:
done.